### PR TITLE
Improved descriptions for "route", "menu", and "header-*" plugins

### LINKED ISSUE #770: Types - amend types and add short description
## This is:

- Feature
- Docs

## Detailed Description
A few of our base types could use some minor amends. Check out `packages/app-security/src/types.ts:44` as an example. The plugins below should follow the same structure.

#### `MenuPlugin` (packages/app-admin/src/types.ts:44)
The type is missing `type: "menu"` property.

Also, add a short description for the type, and [a link](https://docs.webiny.com/docs/webiny-apps/admin/development/plugins-reference/app#menu) to the plugins reference article.

#### `RoutePlugin` (packages/app/src/types.ts:44)
The type is missing `type: "route"` property.

Also, add a short description for the type, and [a link](https://docs.webiny.com/docs/webiny-apps/admin/development/plugins-reference/app#route) to the plugins reference article.

#### `Header` plugins (packages/app-admin/src/types.ts:59)
Apply the same steps to the `HeaderLeftPlugin`, `HeaderRightPlugin`, and `HeaderMiddlePlugin` plugins.

---

With these amends, we are going to improve the developer experience (DX). For example, WebStorm automatically reads the comments, and prints them out in the help dialog (`F1` key while the cursor is positioned on the TS type):

image

### PR DESCRIPTION
**Pull request by drexlar5**
## Related Issue

closes #770

## Your solution

Added `type`, `description` and `links` to plugin documentation to improve the developer experience when using IDEs.

## How Has This Been Tested?

Manual testing

## Screenshots (if relevant):

### CODE REVIEW

File: packages/app-admin/src/types.ts [hunk 1]
```diff
@@ -53,15 +59,33 @@ export type MenuContentSectionPlugin = Plugin & {
     render(props: { Section: typeof Section; Item: typeof Item }): React.ReactNode;
 };
 
+/**
+ * Enables adding custom header elements to the left side of the top bar.
+ * @see https://docs.webiny.com/docs/webiny-apps/admin/development/plugins-reference/app#header-left
+ */
 export type HeaderLeftPlugin = Plugin & {
+    type: "header-left";
+    description: "Enables adding custom header elements to the left side of the top bar.";
```
Comment (doitadrian) [inline packages/app-admin/src/types.ts]: Remove the `description` in the actual TS type.

File: packages/app-admin/src/types.ts [hunk 2]
```diff
@@ -53,15 +59,33 @@ export type MenuContentSectionPlugin = Plugin & {
...
     render(params: RenderParams): React.ReactNode;
 };
 
+/**
+ * Enables adding custom header elements to the right side of the top bar.
+ * @see https://docs.webiny.com/docs/webiny-apps/admin/development/plugins-reference/app#header-right
+ */
 export type HeaderRightPlugin = Plugin & {
+    type: "header-right";
+    description: "Enables adding custom header elements to the right side of the top bar.";
```
Comment (doitadrian) [inline packages/app-admin/src/types.ts]: Remove the `description` in the actual TS type.

File: packages/app-admin/src/types.ts [hunk 3]
```diff
@@ -53,15 +59,33 @@ export type MenuContentSectionPlugin = Plugin & {
...
     render(params: RenderParams): React.ReactNode;
 };
 
+/**
+ * Enables adding custom header elements to the middle of the top bar.
+ * @see https://docs.webiny.com/docs/webiny-apps/admin/development/plugins-reference/app#header-middle
+ */
 export type HeaderMiddlePlugin = Plugin & {
+    type: "header-middle";
+    description: "Enables adding custom header elements to the middle of the top bar.";
```
Comment (doitadrian) [inline packages/app-admin/src/types.ts]: Remove the `description` in the actual TS type.

File: packages/app/src/types.ts [hunk 4]
```diff
@@ -40,6 +40,12 @@ export type ImageComponentPlugin = Plugin & {
     presets: { [key: string]: any };
 };
 
+/**
+ * Enables registering new routes on which you can render your own views.
+ * @see https://docs.webiny.com/docs/webiny-apps/admin/development/plugins-reference/app#route
+ */
 export type RoutePlugin = Plugin & {
+    type: "route";
+    description: "Enables registering new routes on which you can render your own views."
```
Comment (doitadrian) [inline packages/app/src/types.ts]: Remove the `description` in the actual TS type.

File: packages/app/src/types.ts [hunk 5]
```diff
@@ -40,6 +40,12 @@ export type ImageComponentPlugin = Plugin & {
     presets: { [key: string]: any };
 };
 
+/**
+ * Enables registering new routes on which you can render your own views.
```
Comment (doitadrian) [inline packages/app/src/types.ts]: Just `Enables registering new routes.` is enough here.

File: packages/app-admin/src/types.ts (hunk 1 above)
Comment (drexlar5) [reply packages/app-admin/src/types.ts]: The changes has been effected

File: packages/app/src/types.ts (hunk 5 above)
Comment (drexlar5) [reply packages/app/src/types.ts]: This change has been effected.

Comment (doitadrian) [reply packages/app/src/types.ts]: Great! @drexlar5

**Pull request merged**